forc-plugins/forc-doc/src/main.rs

```diff
@@ -121,61 +198,73 @@ pub fn main() -> Result<()> {
             Arc::from(decl_engine),
         ),
         root_attributes,
-        program_kind,
+        typed_program.kind,
         forc_version,
     )?;
 
-    // write contents to outfile
+    // write file contents to doc folder
+    write_content(rendered_docs, doc_path)?;
+    println!("    {}", "Finished".bold().yellow());
+
+    if !build_instructions.no_deps {
+        build_deps(
+            &pkg_manifest.dependencies,
+            manifest.dir(),
+            doc_path,
+            build_instructions,
+        )?;
+    }
+
+    Ok(())
+}
+
+fn build_deps(
+    dependencies: &Option<BTreeMap<String, Dependency>>,
+    manifest_dir: &Path,
+    doc_path: &Path,
+    build_instructions: &Command,
+) -> Result<()> {
+    if let Some(deps) = dependencies {
+        for (dep_name, dep) in deps {
+            if let Dependency::Detailed(dep_details) = dep {
+                if let Some(path) = &dep_details.path {
+                    let manifest_path = manifest_dir.join(path).canonicalize()?;
+                    let dep_manifest = ManifestFile::from_dir(&manifest_path)?;
+                    let dep_pkg_manifest =
+                        if let ManifestFile::Package(pkg_manifest) = &dep_manifest {
+                            pkg_manifest
+                        } else {
+                            bail!("forc-doc does not support workspaces.")
+                        };
+                    build_docs(
+                        &dep_manifest,
+                        dep_pkg_manifest,
+                        doc_path,
+                        build_instructions,
+                    )?;
+                } else {
+                    println!("a path variable was not set for {dep_name}, which is currently the only supported option.")
+                }
+            } else {
+                println!("{dep_name} is a simple format dependency,\nsimple format dependencies don't specify a path to a manfiest file and are unsupported at this time.")
```

Comment:
Should we use bail instead? So this function returns an error, if simple format is used.

Reply:
Since it's for dependencies, I thought it would make more sense to just let the user know but still build the main docs. What do you think?

Reply:
That sounds fine to me!